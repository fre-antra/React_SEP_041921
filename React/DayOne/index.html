<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body>
    <div id="hello-example"></div>
    <script
      crossorigin
      src="https://unpkg.com/react@17/umd/react.development.js"
    ></script>
    <script
      crossorigin
      src="https://unpkg.com/react-dom@17/umd/react-dom.development.js"
    ></script>
    <script>
      class HelloMessage extends React.Component {
        render() {
          return React.createElement('div', null, 'Hello ', this.props.name);
        }
      }

      ReactDOM.render(
        React.createElement(HelloMessage, {
          name: 'Taylor',
        }),
        document.getElementById('hello-example')
      );
    </script>
  </body>
</html>
