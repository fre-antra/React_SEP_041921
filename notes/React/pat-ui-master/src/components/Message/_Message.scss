.msg {
    position: relative;
    display: inline-block;

    @include message-style(
        $msg-background-color, 
        $msg-border-radius, 
        $msg-border,
        $msg-padding,
        $msg-box-width
    );
}

// default Message and different type message
.msg .header {
    font-size: $msg-header-size;
    font-weight: $font-weight-bold;
}

.msg .content {
    display: $msg-pg-display;
    flex-wrap: $msg-pg-display-wrap;
}

.basic-message .content{
    font-size: $msg-pg-size;
}

.list-message .list-header {
    font-size: $msg-header-size;
    font-weight: $font-weight-bold;
    margin-left: $msg-list-margin-left;
}

.list-message .list-content {
    font-size: $msg-pg-size;
}

.icon-message .content {
    font-size: $msg-pg-size;
}

.icon-message .icon {
    float: $msg-icon-float;
    margin: $msg-icon-margin;
}

.dismiss-message .content {
    font-size: $msg-pg-size;
}

.dismiss-message .icon {
    float: $msg-dismiss-float;
}

// different size message
.size-message {
    @include size-message (
        $msg-indent-left
    );
}

.mini-message {
    font-size: $msg-mini-font-size;
}

.tiny-message {
    font-size: $msg-tiny-font-size;
}

.small-message {
    font-size: $msg-small-font-size;
}

.large-message {
    font-size: $msg-large-font-size;
}

.big-message {
    font-size: $msg-big-font-size;
}

.huge-message {
    font-size: $msg-huge-font-size;
}

.massive-message {
    font-size: $msg-massive-font-size;
}

// different color message
.color-message {
    @include color-message(
        $msg-indent-left,
        $msg-pg-size
    );
}

.red-message {
    border-color: $red;
    color: $red;
    background-color: $msg-red-background;
}

.orange-message {
    border-color: $orange;
    color: $orange;
    background-color: $msg-orange-background;
}

.blue-message {
    border-color: $blue;
    color: $blue;
    background-color: $msg-blue-background;
}

.pink-message {
    border-color: $pink;
    color: $pink;
    background-color: $msg-pink-background;
}

.black-message {
    border-color: $black;
    color: $white;
    background-color: $black;
}