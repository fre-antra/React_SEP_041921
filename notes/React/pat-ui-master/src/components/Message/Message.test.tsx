import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import Message ,{ IMessagesProps } from './Message';
import { Icon } from '../..';

describe('Message', () => {
    it('should match snapshot', () => {
        const {asFragment} = render(<Message> Snapshot Message </Message>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render default message', () => {
        // Default message
        const msgBasicProps: IMessagesProps = {
            msgType: 'basic',
            msgSize: 'large',
            msgColor: 'white',
            className: 'test'
        }
        const msgBasicWrapper = render(<Message {...msgBasicProps}></Message>);
        const msgElement = screen.getByTestId('message-element');
        expect(msgElement).toHaveClass('msg msg-basic msg-large msg-white test');
    });

    it('should render message with different props', () => {
        // List message
        const msgListProps: IMessagesProps = {
            msgType: 'list',
            msgSize: 'large',
            msgColor: 'white',
            className: 'test',
            msgHeader: 'List Message'
        }
        const msgListWrapper = render(<Message {...msgListProps}></Message>);
        const msgListElement = msgListWrapper.queryByText('List Message');
        expect(msgListElement).toBeInTheDocument();
        const listElement = screen.queryByTestId('message-element');
        expect(listElement).toHaveClass('msg msg-list msg-large msg-white test');

        // Icon message
        const msgIconProps: IMessagesProps = {
            msgType: 'icon',
            msgSize: 'large',
            msgColor: 'white',
            className: 'test',
            msgHeader: 'Icon Message'
        }
        const msgIconWrapper = render(<Message {...msgIconProps}></Message>);
        const msgIconElement = msgIconWrapper.queryByText('Icon Message');
        expect(msgIconElement).toBeInTheDocument();

        // Dismiss message
        const msgDismissProps: IMessagesProps = {
            msgType: 'dismiss',
            msgSize: 'large',
            msgColor: 'white',
            msgHeader: 'Dismiss Message',
            msgOnClick: jest.fn()
        }
        const msgDismissWrapper = render(<Message></Message>);
        const dismissElement = screen.queryAllByTestId('message-element');
        expect(msgDismissProps.msgOnClick).toHaveBeenCalledTimes(0);
    });

});