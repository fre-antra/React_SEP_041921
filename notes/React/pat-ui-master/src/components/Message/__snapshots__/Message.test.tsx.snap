// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="msg msg-basic msg-large msg-white"
    data-testid="message-element"
  >
    <div
      class="icon"
      data-testid="icon-element"
    >
      <p />
    </div>
    <div
      class="header"
    >
      <p />
    </div>
    <div
      class="content"
    >
      <p />
    </div>
  </div>
</DocumentFragment>
`;
